packages/contracts/contracts/L2/messaging/L2StandardERC721Factory.sol

```diff
@@ -0,0 +1,51 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+
+/* Contract Imports */
+import { L2StandardERC721 } from "../../standards/L2StandardERC721.sol";
+
+/**
+ * @title L2StandardERC721Factory
+ * @dev Factory contract for creating standard L2 ERC721 representations of L1 ERC721s
+ * compatible with and working on the NFT bridge.
+ */
+contract L2StandardERC721Factory {
+    event StandardL2ERC721Created(address indexed _l1Token, address indexed _l2Token);
+
+    // Address of the L2 ERC721 Bridge. It currently exists only on Optimistic Kovan.
+    address public constant L2_ERC721_BRIDGE = 0xA779A0cA89556A9dffD47527F0aad1c2e0d66e46;
+
+    // Maps an L2 ERC721 token address to a boolean that returns true if the token was created
+    // with the L2StandardERC721Factory.
+    mapping(address => bool) public isStandardERC721;
+
+    // Maps an L1 ERC721 token address to an L2 Standard ERC721 token address. This mapping can
+    // only be updated once per L1 ERC721 token.
+    mapping(address => address) public standardERC721Mapping;
```

Comment:
Same comment as above - I feel like an offchain tokenlist is better for this because using this factory is permissionless so its not the best indicator of legitimacy

Reply:
The purpose of this mapping isn't actually to maintain an on-chain token list. The mapping is there to ensure that there's one, and only one, L2 Standard ERC721 contract for each L1 ERC721 contract (which is enforced [here](https://github.com/ethereum-optimism/optimism/pull/2577/files/e307559010d67a4cc8449ebdf081a682cca5a5ab#diff-d1c461b922bbc3893e80ce8fb6371f7955bd47914f604debaf5ac1633b6a9011R40)). Since these standard contracts implement the exact same functionality, there's no need to have more than one for each L1 contract. Also, this mapping makes it easy for frontends to find out which L2 Standard ERC721 corresponds to an L1 ERC721, which makes it easy to know which L2 contract to bridge to.